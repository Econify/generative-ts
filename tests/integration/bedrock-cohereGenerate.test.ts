import {
  CohereGenerateApi,
  createAwsBedrockModelProvider,
} from "@packages/core";

test("Bedrock - Cohere Generate", async () => {
  // arrange
  const cohereGenerate = createAwsBedrockModelProvider({
    api: CohereGenerateApi,
    modelId: "cohere.command-text-v14",
  });

  // act
  const response = await cohereGenerate.sendRequest({
    prompt: "Brief history of NY Mets:",
    max_tokens: 100,
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
