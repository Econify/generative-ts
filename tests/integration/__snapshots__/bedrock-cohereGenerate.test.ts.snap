// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bedrock - Cohere Generate 1`] = `
{
  "generations": [
    {
      "finish_reason": "string",
      "id": "string",
      "text": "string",
    },
  ],
  "id": "string",
  "prompt": "string",
}
`;
